apps/andi/assets/js/app.js

```diff
@@ -35,15 +35,17 @@ Hooks.showSnackbar = {
 Hooks.readFile = {
     mounted() {
         this.el.addEventListener("change", e => {
-            var file = this.el.files[0];
             this.pushEvent("file_upload_started");
+            var file = this.el.files[0];
 
             fileToText(this.el.files[0]).then(fileAsText => {
                 this.pushEvent("file_upload", {
                     file: fileAsText,
                     fileType: file["type"],
                     fileSize: file["size"]
                 });
+            }, reason => {
+                reason == "cancel" & this.pushEvent("file_upload_cancelled");
```

Comment:
wait, what does `&` do here? Is it an XOR?

Reply:
no its basically a shorthand if statement

Reply:
yeah that should be `&&`